clang/test/AST/ast-print-language-linkage.cpp

```diff
@@ -0,0 +1,27 @@
+// RUN: %clang_cc1 -ast-print %s -o - | FileCheck %s
+
+// CHECK: extern "C" int printf(const char *, ...);
+extern "C" int printf(const char *...);
+
+// CHECK: extern "C++" {
+// CHECK-NEXT:   int f(int);
+// CHECK-NEXT:   int g(int);
+// CHECK-NEXT: }
+extern "C++" int f(int), g(int);
+
+// CHECK: extern "C" {
+// CHECK-NEXT:  void foo();
+// CHECK-NEXT:  int x;
+// CHECK-NEXT:  int y;
+// CHECK-NEXT: }
+extern "C" {
+  void foo(void);
+  int x, y;
```

Comment:
This part in particular is unfortunate.  What do we look like in non-linkage specs?

Reply:
If you mean the splitting of `int x, y` into `int x; int y`, that's actually the «default». The declarators are split into different declarations whenever it's possible. So, the splitting doesn't happen only when there is a definition of a tagged type within the declaration, I believe. So, for example:

`union {} a, b;` → `union {\n} a, b;`

but

`int a, b;` → `int a; int b;`


That fact, that this very splitting already occurs in DeclPrinter and leads to the deviation from the original source, was the reason why I felt more comfortable to go forward with adding braces.

Reply:
I see, yeah, I think this is fine then.  Same with the functions, but I'll give Aaron a chance to comment.

Reply:
I like the new version.  Not quite perfect fidelity, but at least matches what is going on.